### PR TITLE
EUI-4166 Task event completion

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4166
https://tools.hmcts.net/jira/browse/EUI-4167
https://tools.hmcts.net/jira/browse/EUI-4168

### Change description ###
Task completion - Task check at event completion - task cancelled or completed
Task completion - Task check at event completion - task has been assigned to another user
Task completion - Task check at event completion - task became unassigned

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/shared/components/case-editor/services/event-completion-state-machine.service.ts [hunk 1]
```diff
@@ -80,22 +84,26 @@ export class EventCompletionStateMachineService {
       if (taskPayLoad.task_required_for_event) {
         const task = taskPayLoad.tasks.find(x => x.id === context.task.id);
         if (task) {
```
Comment (udaydenduluri33) [inline src/shared/components/case-editor/services/event-completion-state-machine.service.ts:86]:
```suggestion
        if (task && task.task_state) {
```

File: src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-cancelled/case-event-completion-task-cancelled.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    fixture.detectChanges();
+  });
+
+  it('should display error message task cancelled', () => {
+    const heading: DebugElement = fixture.debugElement.query(By.css('.govuk-heading-m'))
+    const headingHtml = heading.nativeElement as HTMLElement;
+    expect(headingHtml.innerText).toBe('Task cancelled/marked as done');
+  });
+
+  it('shoud emit event can be completed true when clicked on continue button', () => {
```
Comment (connorpgpmcelroy) [inline src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-cancelled/case-event-completion-task-cancelled.component.spec.ts:51]: minor comment - shoud -> should

**Review by udaydenduluri33 — APPROVED** [review]
@johnbenjamin-hmcts thanks for the changes

File: src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    this.caseId = this.route.snapshot.params['cid'];
+  }
+
+  public ngOnInit(): void {
+    const caseworkers: any = JSON.parse(this.sessionStorageService.getItem('caseworkers'));
+    const caseworker = caseworkers.find(x => x.idamId === this.assignedUserId);
+    this.assignedUserName = caseworker !== undefined ? `${caseworker.firstName} ${caseworker.lastName}` : 'another user';
+    // TODO: If the task is not assigned to a caseworker, then
+    // we have to perform an api call to check whether the task is assigned to judicial user
+    // and display the judicial user name instead of 'another user'
```
Comment (connorpgpmcelroy) [inline src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts:34]: Minor query - Presumably this is work yet to be done that is not required for this PR?

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM apart from few comments

**Review by udaydenduluri33 — APPROVED** [review]

**Pull request merged**